src/Buckle/Buckle/CodeAnalysis/Binding/Binder.cs

```diff
@@ -667,17 +667,10 @@ private struct BinderState {
                 }
             }
 
-            if (symbols.Length == 1) {
-                var failed = false;
-
-                if (diagnostics.FilterOut(DiagnosticType.Warning).Any())
-                    failed = true;
-
+            if (symbols.Length == 1 && diagnostics.FilterOut(DiagnosticType.Warning).Any()) {
```

Comment:
The BindCallExpression was updated the diagnostics prematurely, it was supposed to only updated the diagnostics when returning from the function, and there was an edge case here where it would update them too early if only 1 overload to a function was found